### DISCUSSION TITLE
Changing the amount of entries while rofi is running in a custom plugin

### DISCUSSION THREAD

**Discussion by LordMZTE** [davatorium/rofi#1654] [Q&A] [answered]
I'm writing a rofi plugin which loads quite a bit of data, so it would make rofi very slow to start. Is there a way that I can add my entries afterwards? rofi only seems to call `_get_num_entries` once on startup, which obviously returns 0 since my data isn't loaded yet. Later it'll be loaded by a glib async callback though, but rofi doesn't seem to respect this as it never looks for more entries again. Do I have to notify rofi about the new entries?

**Comment by DaveDavenport** [comment, marked as answer]
on a side note: glib async is very slow (I hit this in dmenu mode), you might want to consider a custom implementation.

To force rofi to reload, you can call `rofi_view_reload()`. 
See  the dmenu implementation: https://github.com/davatorium/rofi/blob/next/source/modes/dmenu.c#L216

**Reply by LordMZTE** [reply]
Thanks! This fixes my issue. Note that I had to declare the function myself, though. The `view.h` header doesn't seem to be included in the rofi package on Arch Linux.

**Reply by DaveDavenport** [reply]
Yeah, cleaning up headers for plugins is on my todo, but I don't have a lot of time.
